Build failure on MSYS2: error cast from 'HINSTANCE' {aka 'HINSTANCE__*'} to 'int' loses precision [-fpermissive]

Enigma version: 1.30 (git 2d10fcaa).
MSYS2 environment used to build: MINGW64
g++ version:
```
g++ --version
g++.exe (Rev7, Built by MSYS2 project) 10.3.0
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```
Compilation log:
```
Making all in m4
make[1]: Entering directory '/d/Linux_home/nathan/src/enigma/m4'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/d/Linux_home/nathan/src/enigma/m4'
Making all in lib-src
make[1]: Entering directory '/d/Linux_home/nathan/src/enigma/lib-src'
Making all in oxydlib
make[2]: Entering directory '/d/Linux_home/nathan/src/enigma/lib-src/oxydlib'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/d/Linux_home/nathan/src/enigma/lib-src/oxydlib'
Making all in lua
make[2]: Entering directory '/d/Linux_home/nathan/src/enigma/lib-src/lua'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/d/Linux_home/nathan/src/enigma/lib-src/lua'
Making all in zipios++
make[2]: Entering directory '/d/Linux_home/nathan/src/enigma/lib-src/zipios++'
Making all in zipios++
make[3]: Entering directory '/d/Linux_home/nathan/src/enigma/lib-src/zipios++/zipios++'
make  all-am
make[4]: Entering directory '/d/Linux_home/nathan/src/enigma/lib-src/zipios++/zipios++'
make[4]: Leaving directory '/d/Linux_home/nathan/src/enigma/lib-src/zipios++/zipios++'
make[3]: Leaving directory '/d/Linux_home/nathan/src/enigma/lib-src/zipios++/zipios++'
Making all in src
make[3]: Entering directory '/d/Linux_home/nathan/src/enigma/lib-src/zipios++/src'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/d/Linux_home/nathan/src/enigma/lib-src/zipios++/src'
make[3]: Entering directory '/d/Linux_home/nathan/src/enigma/lib-src/zipios++'
make[3]: Nothing to be done for 'all-am'.
make[3]: Leaving directory '/d/Linux_home/nathan/src/enigma/lib-src/zipios++'
make[2]: Leaving directory '/d/Linux_home/nathan/src/enigma/lib-src/zipios++'
Making all in tinygettext
make[2]: Entering directory '/d/Linux_home/nathan/src/enigma/lib-src/tinygettext'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/d/Linux_home/nathan/src/enigma/lib-src/tinygettext'
Making all in enigma-core
make[2]: Entering directory '/d/Linux_home/nathan/src/enigma/lib-src/enigma-core'
g++ -std=c++14 -DHAVE_CONFIG_H -I. -I../../src    -ID:/msys64/mingw64/include/SDL2 -Dmain=SDL_main  -g -O2 -DENABLE_ASSERT -g -DCXXLUA -MT ecl_system.o -MD -MP -MF .deps/ecl_system.Tpo -c -o ecl_system.o ecl_system.cc
ecl_system.cc: In function 'bool ecl::BrowseUrl(std::string)':
ecl_system.cc:122:16: error: cast from 'HINSTANCE' {aka 'HINSTANCE__*'} to 'int' loses precision [-fpermissive]
  122 |     result == ((int)ShellExecute(NULL, "open", url.c_str(), NULL, NULL, SW_SHOWNORMAL) >= 32);
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ecl_system.cc: In function 'bool ecl::ExploreFolder(std::string)':
ecl_system.cc:144:16: error: cast from 'HINSTANCE' {aka 'HINSTANCE__*'} to 'int' loses precision [-fpermissive]
  144 |     result == ((int)ShellExecute(NULL, "explore", path.c_str(), NULL, NULL, SW_SHOWNORMAL) >= 32);
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [Makefile:468: ecl_system.o] Error 1
make[2]: Leaving directory '/d/Linux_home/nathan/src/enigma/lib-src/enigma-core'
make[1]: *** [Makefile:390: all-recursive] Error 1
make[1]: Leaving directory '/d/Linux_home/nathan/src/enigma/lib-src'
make: *** [Makefile:477: all-recursive] Error 1
```

--- a/lib-src/enigma-core/ecl_system.cc
+++ b/lib-src/enigma-core/ecl_system.cc
@@ -119,7 +119,7 @@ bool ecl::FolderCreate(const std::string &fname) {
 bool ecl::BrowseUrl(const std::string url) {
     bool result = true;
 #ifdef __MINGW32__
-    result == ((int)ShellExecute(NULL, "open", url.c_str(), NULL, NULL, SW_SHOWNORMAL) >= 32);
+    result == ((INT_PTR)ShellExecute(NULL, "open", url.c_str(), NULL, NULL, SW_SHOWNORMAL) >= 32);
 #elif MACOSX
     CFStringRef cfurlStr = CFStringCreateWithCString(NULL, url.c_str(), kCFStringEncodingASCII);
 
@@ -141,7 +141,7 @@ bool ecl::BrowseUrl(const std::string url) {
 bool ecl::ExploreFolder(const std::string path) {
     bool result = true;
 #ifdef __MINGW32__
-    result == ((int)ShellExecute(NULL, "explore", path.c_str(), NULL, NULL, SW_SHOWNORMAL) >= 32);
+    result == ((INT_PTR)ShellExecute(NULL, "explore", path.c_str(), NULL, NULL, SW_SHOWNORMAL) >= 32);
 #elif MACOSX
     FSRef fref;
     FSPathMakeRef((UInt8 *)path.c_str(), &fref, NULL);

